src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy

```diff
@@ -291,4 +304,31 @@ class IntelliJPlugin implements Plugin<Project> {
         configurations.getByName(JavaPlugin.COMPILE_CONFIGURATION_NAME).getAllDependencies().each closure
 
     }
+
+    private static def downloadExternalPlugin(@NotNull Project project, @NotNull Map<String, String> plugin) {
+        def host = "http://plugins.jetbrains.com"
+
+        new File("$project.projectDir/externalPluginsLibs/").mkdirs()
+
+        new AntBuilder().get(src:"$host/plugin/download?pluginId=$plugin.id&version=$plugin.version", dest:"$project.projectDir/externalPluginsLibs/$plugin.id.$plugin.type", skipexisting:true)
+
+        return extractExternalPluginJars(project, plugin)
+    }
+
+    private static def extractExternalPluginJars(@NotNull Project project, @NotNull Map<String, String> plugin) {
+        if ("jar".equals(plugin.type)) {
+            return [new File("$project.projectDir/externalPluginsLibs/$plugin.id.$plugin.type")]
+        }
+
+        new AntBuilder().unzip(src: "$project.projectDir/externalPluginsLibs/$plugin.id.$plugin.type", dest: "$project.projectDir/externalPluginsLibs/", overwrite: "false")
+
+        def libsDir = new File("$project.projectDir/externalPluginsLibs/$plugin.unzipTarget/lib")
+
+        return libsDir.listFiles(new FilenameFilter() {
+            @Override
+            boolean accept(File dir, String name) {
+                return name.endsWith("jar");
+            }
+        });
+    }
```

Comment:
This code downloads and unpacks plugin on each build. I really don't like this. Hope to find a way to install downloaded zips and jars to local ivy repository.